### PR TITLE
perf improvments on getRepos and API change on gitHub*

### PR DESCRIPTION
**Pull request by jrit**
~80% perf improvement on repo lists with use of parallel.

[semver-major] API change to all `gitHub` methods.

Tests are updated to required use of cvr as the test target since other
part of the tests already assumed a specific repo but it was hidden in
env vars

@dmaloneycalu please have a look

Ref: https://github.com/vokal/cvr-view/issues/135

### CODE REVIEW

File: source/github.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+        {
+            return r.config.url === hookUrl;
+        } )[ 0 ];
+
+        if( existingHook )
+        {
+            return done( null, existingHook );
+        }
+
+        return done( null, null );
```
Comment (dmaloneycalu) [inline source/github.js]: could combine with above `if` as `return done( null, existingHook || null );`

File: source/github.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+api.getHookByUrl = function ( accessToken, owner, repoName, hookUrl, done )
+{
+    authenticate( accessToken );
+
+    var onHooks = function ( err, res )
+    {
+        if( err )
+        {
+            return done ( err );
```
Comment (dmaloneycalu) [inline source/github.js]: no space after `done`

File: source/github.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+
+api.deleteHook = function ( accessToken, owner, repoName, hookUrl, done )
+{
+    authenticate( accessToken );
+
+    var onHook = function ( err, existingHook )
+    {
+        if( err )
+        {
+            return done ( err );
```
Comment (dmaloneycalu) [inline source/github.js]: no space again

File: source/github.js (hunk 1 above)
Comment (jrit) [inline source/github.js]: This is code that I just moved from another file, but this is a really weird callback, not sure why I wrote it like this when I did, so thanks for noticing.

**Comment by dmaloneycalu** [comment]
Couple of spaces, otherwise LGTM. Since this is a major change, want anyone else to take a look at it?

**Comment by jrit** [comment]
I think your eyes are enough, none of the tests really changed, and it is only a major bump because I felt the existing API was ugly. If I'd staged it better and didn't move everything around at once the diff would appear much smaller.

**Comment by dmaloneycalu** [comment]
This could save a major version update if you add a map from the old api names to the new ones. I presume this major change here would require a change for cvr-view as well.

**Comment by jrit** [comment]
Correct, but at this stage it is an easy change and preserving the old API is not worth the current or future maintenance.

**Comment by dmaloneycalu** [comment]
Alrighty :+1:

**Pull request merged**